code/FulpstationCode/fulp_medborg/fulp_medborg_updates.dm

```diff
@@ -0,0 +1,105 @@
+/datum/design/borg_upgrade_medbeam
+	name = "Cyborg Upgrade (Heal Beam)"
+	id = "borg_upgrade_medbeam"
+	build_type = MECHFAB
+	build_path = /obj/item/borg/upgrade/medbeam
+	materials = list(/datum/material/iron = 10000, /datum/material/glass = 10000, /datum/material/plasma = 4000, /datum/material/uranium = 4000)
+	construction_time = 80
+	category = list("Cyborg Upgrade Modules")
+
+/obj/item/borg/upgrade/medbeam
+	name = "medical cyborg heal beam"
+	desc = "An upgrade to the Medical module, installing a built-in \
+		healing beam."
+	icon_state = "cyborg_upgrade3"
+	require_module = 1
+	module_type = list(/obj/item/robot_module/medical)
+	var/backpack = FALSE //True if we get the defib from a physical backpack unit rather than an upgrade card, so that we can return that upon deactivate()
+
+/obj/item/borg/upgrade/medbeam/action(mob/living/silicon/robot/R, user = usr)
+	. = ..()
+	if(.)
+		var/obj/item/gun/medbeam/cyborg/MB = new(R.module)
+		R.module.basic_modules += MB
+		R.module.add_module(MB, FALSE, TRUE)
+
+/obj/item/borg/upgrade/medbeam/deactivate(mob/living/silicon/robot/R, user = usr)
+	. = ..()
+	if (.)
+		var/obj/item/gun/medbeam/cyborg/MB = locate() in R.module
+		R.module.remove_module(MB, TRUE)
+
+/obj/item/gun/medbeam/cyborg
+	name = "Integrated Medical Beamgun"
+	desc = "Advanced protonic nano-something or other miracle healing beam. Crossing its stream with another is ill-advised."
+	var/power_cost = 75
+
+/obj/item/gun/medbeam/cyborg/process_fire(atom/target, mob/living/silicon/robot/user, message = TRUE, params = null, zone_override = "", bonus_spread = 0)
+	if(user.cell.charge < power_cost) //Check if we have enough power.
+		to_chat(user, "<span class='warning'>Power inadequate to initiate beam projection.</span>")
+		LoseTarget()
+		return
+
+	playsound(target, 'sound/effects/magic.ogg', 10, TRUE, TRUE)
+	. = ..()
+
+/obj/item/gun/medbeam/cyborg/on_beam_tick(mob/living/target)

```

Comment:
can't you just... use the syndicate medical cyborg one?

Reply:
No. It's obviously half-assed as hell and doesn't drain power at all.